src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -59,7 +59,7 @@ public string Charset
                     // Remove charset parameter
                     if (charsetParameter != null)
                     {
-                        _parameters.Remove(charsetParameter);
+                        Parameters.Remove(charsetParameter);
```

Comment:
It's intended to avoid allocating the `Parameters` collection unless we need to mutate it.